src/ext/respToRedisLoader.c

```diff
@@ -324,18 +332,35 @@ _LIBRDB_API void RDBX_setPipelineDepth(RdbxRespToRedisLoader *r2r, int depth) {
     r2r->pendingCmds.pipelineDepth = (depth <= 0 || depth>PIPELINE_DEPTH_MAX) ? PIPELINE_DEPTH_DEF : depth;
 }
 
+/* Create a loader from an existing file descriptor */
 _LIBRDB_API RdbxRespToRedisLoader *RDBX_createRespToRedisFd(RdbParser *p,
                                                             RdbxToResp *rdbToResp,
                                                             RdbxRedisAuth *auth,
                                                             int fd) {
-    RdbxRespToRedisLoader *ctx;
-    if ((ctx = RDB_alloc(p, sizeof(RdbxRespToRedisLoader))) == NULL) {
-        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP_FAILED_ALLOC,
-                        "Failed to allocate struct RdbxRespToRedisLoader");
+    /* Ensure the socket is in blocking mode */
+    int flags = fcntl(fd, F_GETFL, 0);
+    if (flags == -1 || fcntl(fd, F_SETFL, flags & ~O_NONBLOCK) == -1) {
```

Comment:
RDBX_createRespToRedisFd() is a public API right? It accepts an fd, then changes some of its socket configs. Now, user has a non-blocking socket with a different timeout but the user is not aware of. I guess it is better to revert when we are done with the fd or at least document it?

Reply:
revert. thanks.